ts-tests/tests/precompile-contract.test.ts

```diff
@@ -0,0 +1,233 @@
+        );
+
+        // skip test if already delegated
+        if (balance.reserved.toNumber() === 0) {
+            await executeTransaction(delegateWithAutoCompound, 'delegateWithAutoCompound');
+            const { data: afterDelegateBalance } = await context.api.query.system.account(evmAccountRaw.mappedAddress);
+
+            expect(balance.free.toNumber() - toBigNumber(60)).to.closeTo(
+                afterDelegateBalance.free.toNumber(),
+                toBigNumber(1)
+            );
+            expect(afterDelegateBalance.reserved.toNumber()).to.eq(toBigNumber(60));
+            const collator = await collatorDetails();
+            expect(collator.value).to.eq(autoCompoundPercent);
+
+            balance = afterDelegateBalance;
+        }
+
+        // delegatorBondMore(collator, amount)
+        const delegatorBondMore = precompileContract.methods.delegatorBondMore(collatorPublicKey, toBigNumber(1));
+        await executeTransaction(delegatorBondMore, 'delegatorBondMore');
+
+        const { data: balanceAfterBondMore } = await context.api.query.system.account(evmAccountRaw.mappedAddress);
+        expect(balanceAfterBondMore.free.toNumber()).to.closeTo(
+            balanceAfterBondMore.free.toNumber() - toBigNumber(1),
+            toBigNumber(1)
+        );
+        expect(balanceAfterBondMore.reserved.toNumber()).to.eq(balance.reserved.toNumber() + toBigNumber(1));
+
+        // setAutoCompound(collator, percent);
+        const setAutoCompound = precompileContract.methods.setAutoCompound(collatorPublicKey, autoCompoundPercent + 5);
+        await executeTransaction(setAutoCompound, 'setAutoCompound');
+        const collatorAfterCompound = await collatorDetails();
+        expect(collatorAfterCompound.value).to.eq(autoCompoundPercent + 5);
+
+        // scheduleDelegatorBondLess(collator, amount)
+        expect(await isPendingRequest()).to.be.false;
+        const scheduleDelegatorBondLess = precompileContract.methods.scheduleDelegatorBondLess(
+            collatorPublicKey,
+            toBigNumber(5)
+        );
+        await executeTransaction(scheduleDelegatorBondLess, 'scheduleDelegatorBondLess');
+        expect(await isPendingRequest()).to.be.true;
+
+        // cancelDelegationRequest(collator)
+        const cancelDelegationRequest = precompileContract.methods.cancelDelegationRequest(collatorPublicKey);
+        expect(await isPendingRequest()).to.be.true;
+        await executeTransaction(cancelDelegationRequest, 'cancelDelegationRequest');
+        expect(await isPendingRequest()).to.be.false;
+
+        // In case evm is not enabled in Normal Mode, switch back to filterMode, after test.
+        let extrinsic = context.api.tx.sudo.sudo(context.api.tx.extrinsicFilter.setMode(filterMode));
+        await signAndSend(extrinsic, context.alice);
+    });
+
+    // once we add ability to customise minimum waiting time for `executeDelegationRequest` implement tests for the rest functions
```

Comment:
Created dedicated task for that [P-360 : Add the rest tests for missing functions](https://linear.app/litentry/issue/P-360/add-the-rest-tests-for-missing-functions)